### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by shubham14p3**
Created our first bubble sort project in ruby
We had two modules as per requirement
#bubble_sort
#bubble_sort_by

Inputs for the same are as follow :
a. #bubble_sort
array = [5,9,7,3,1]
bubble_sort(array)

b. #bubble_sort_by
array = ["hi","hello","hey"]
bubble_sort_by(array) do |left,right|
right.length - left.length
end

Output for the above were as follow:-

a. #bubble_sort
This is the array that needs to be sorted : [5, 9, 7, 3, 1]
[1, 3, 5, 7, 9]

b. #bubble_sort_by
This is the array that needs to be sorted : ["hi", "hello", "hey"]
["hi", "hey", "hello"]

### CODE REVIEW

**Comment by shubham14p3** [comment]
The earlier Repo &  Code was deleted and closed, as we were unable to enable linters 
Previous Pull Request : .https://github.com/shubham14p3/Advanced-Building-Blocks-Bubble-Sort/pull/2
Changes were made and have been updated for the same.

File: sort_by.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    swapped = 1
+    while swapped < passes
+      if yield(array[swapped - 1], array[swapped]).negative?
+        array[swapped - 1], array[swapped] = array[swapped], array[swapped - 1]
+      end
+      swapped += 1
+    end
+    passes -= 1
+  end
+  puts array.to_s
```
Comment (kobiyoyo) [inline sort_by.rb]: i recommend you remove the `puts` and `to_s ` .it should be like this `array`

File: sort_by.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  while swapped
+    swapped = false
+    (n - 1).times do |i|
+      if array[i] > array[i + 1]
+        array[i], array[i + 1] = array[i + 1], array[i]
+        swapped = true
+      end
+    end
+  end
+  puts array.to_s
```
Comment (kobiyoyo) [inline sort_by.rb]: i recommend you remove the `puts` and `to_s ` .it should be like this `array`

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team   
#### Great Job so far, you are almost there .
### General


- [ ] `bubble_sort_by`
 `bubble_sort_by(["hi","hello","hey"]) do |left,right|`
    `left.length - right.length`
  `end`
 ` => should return ["hi", "hey", "hello"]`
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: sort_by.rb (hunk 1 above)
Comment (shubham14p3) [reply sort_by.rb]: Made the changes

File: sort_by.rb (hunk 2 above)
Comment (shubham14p3) [reply sort_by.rb]: Made the changes

**Comment by shubham14p3** [comment]
Kindly let me know if any thing else is required.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello, Thank you for the improvements you've done.
However, there is still a bug. Take a look at this execution
```ruby
2.6.5 :002 > bubble_sort_by(["hi","hello","hey"]) do |left,right|
2.6.5 :003 >     left.length - right.length
2.6.5 :004?>   end
 => ["hello", "hey", "hi"] 
2.6.5 :005 >
```
According to the [project description](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks), the result must be 
```ruby
=> ["hi", "hey", "hello"]
```

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by shubham14p3** [comment]
Error Fixed

**Comment by majovanilla** [comment]
Hey @shubham14p3 

## You did a *great* job on this project! 💜 

Your bubble_sort meets all the requirements asked by previous TSE's.
Your pull request message is perfect! :fire: 

## You are approved to merge

Happy coding 🤓

--Maria Reyes

**Comment by shubham14p3** [comment]
Changes Made & Code approved,
Thanks

**Pull request merged**